### PR TITLE
feat hub/vendor/customer.py: added support for sending reactivation event

### PR DESCRIPTION
**Pull request by jackiemunroe**
Added support to send subscription reactivation event data to Salesforce/Basket
Refactored StripeCustomerSubscriptionUpdated class
Improved code coverage

https://jira.mozilla.com/browse/COPS-815
https://github.com/mozilla/subhub/issues/305
https://github.com/mozmeao/basket/issues/362

### CODE REVIEW

File: src/hub/vendor/customer.py [hunk 1]
```diff
@@ -257,159 +256,224 @@ def run(self) -> None:
...
-                self.get_latest_invoice(invoice_id)
-            else:
+            latest_invoice = stripe.Invoice.retrieve(id=invoice_id)
+            logger.info("get latest invoice", latest_invoice=latest_invoice)
+
+            if latest_invoice.charge is not None and latest_invoice.number is not None:
                 return latest_invoice
+
+            attempts -= 1
+            time.sleep(0.2)
```
Comment (marty331) [inline src/hub/vendor/customer.py]: In get_latest_invoice() you are implementing exponential backoff, you should use shared/vendor - retrieve_stripe_invoice.

File: src/hub/vendor/customer.py [hunk 2]
```diff
@@ -257,159 +256,224 @@ def run(self) -> None:
...
+                latest_charge.payment_method_details.card.last4 is not None
+                or latest_charge.payment_method_details.card.brand is not None
             ):
-                time.sleep(0.2)
-                self.get_latest_charge(charge)
-            else:
                 return latest_charge
+
+            attempts -= 1
+            time.sleep(0.2)
```
Comment (marty331) [inline src/hub/vendor/customer.py]: Same as above, should use shared/vendor - retrieve_stripe_charge ln 346

File: src/hub/vendor/customer.py (hunk 1 above)
Comment (jackiemunroe) [reply src/hub/vendor/customer.py]: removed method and updated references to use shared/vendor method

File: src/hub/vendor/customer.py (hunk 2 above)
Comment (jackiemunroe) [reply src/hub/vendor/customer.py]: removed method and updated references to use shared/vendor method

**Pull request merged**